### PR TITLE
Author show top review #13

### LINKED ISSUE #13: User Story 15, Author Show Page displays top review for each book
```
As a Visitor,
When I visit an author's show page,
For each book written by that author
I should see one of the highest rated reviews
(review should contain the title, score, and user name)
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #15

child of #33

depends on #14

depends on #16

### PR DESCRIPTION
**Pull request by Vjp888**
Adds top reviews to author show page.

It will successfully add info about the top review for each book on the author's show page
preemptively added the link to the user show page and expect in the test, they will need to be uncommented when merged as the test crashed when set without appropriate routing.

All tests passing otherwise

closes #13

### CODE REVIEW

File: spec/features/author_show_spec.rb [hunk 1]
```diff
@@ -75,4 +75,37 @@
...
+
+    visit author_path(author)
+
+    within "#book-#{book.id}" do
+      expect(page).to have_content("Top Review:")
+      expect(page).to have_content("Rating: 5 out of 5")
+      expect(page).to have_content("Title: meh")
+      expect(page).to have_content("User: bob")
+      expect(page).to have_link("bob")
+      click_link 'bob'
```
Comment (milevy1) [inline spec/features/author_show_spec.rb]: What is this line 99 click_link 'bob' for?

File: spec/features/author_show_spec.rb [hunk 2]
```diff
@@ -75,4 +75,37 @@
...
+    end
+    within "#book-#{book_2.id}" do
+      expect(page).to have_content("Top Review:")
+      expect(page).to have_content("Rating: 3 out of 5")
+      expect(page).to have_content("Title: hammy")
+      expect(page).to have_content("User: harbi")
+      expect(page).to have_link("harbi")
+    end
+
+    # expect(current_path).to eq(user_index_path) Renable when merged
```
Comment (milevy1) [inline spec/features/author_show_spec.rb]: Will the user_index_path need an argument of the user?

File: spec/features/author_show_spec.rb (hunk 1 above)
Comment (Vjp888) [reply spec/features/author_show_spec.rb]: That was placed as a quick check for the user show links working. But I added a second book to make sure it was populating correctly and didn't move it. I moved it to sit below the withins and before the updated path spec so it will not break the other tests.

Comment (milevy1) [reply spec/features/author_show_spec.rb]: Ok, that makes sense now.

**Comment by milevy1** [comment]
Looks good, I approve to merge.
